components/SendMoneyToCollectiveBtn.js

```diff
@@ -117,22 +128,39 @@ const addPaymentMethodsData = graphql(paymentMethodsQuery, {
   },
 });
 
-const sendMoneyToCollectiveMutation = gql`
-  mutation SendMoneyToCollective($order: OrderInputType!) {
-    createOrder(order: $order) {
+const collectiveBalanceQuery = gql`
```

Comment:
I don't think it's a good approach to create a new "fake" query here, can you think about an alternative?

Reply:
@znarf : We use the same approach in https://github.com/opencollective/opencollective-frontend/blob/a1be7711930ce2e9bf644a39af13ea705a1caf78/components/changelog/ChangelogTrigger.js#L57. Off the top of my head, one thing we can try is try re-using a query that we already have, such as, [`collectivePageQuery`](https://github.com/opencollective/opencollective-frontend/blob/7fa0a25fcacf8305271b6c56feaa1e7d51b7a2f3/components/collective-page/graphql/queries.js#L11). 

But this approach I didn't quite prefer, as one problem with that is, when some fields are missing data Apollo returns null for the `readQuery` call. According to the docs: 

>If the cache is missing data for any of the query's fields, readQuery returns null. It does not attempt to fetch data from your GraphQL server.

I've run into this problem before. Also another thing is that we only need the `balance` and reusing a query might return all the fields (especially in the case of `collectivePageQuery`) it's large and therefore I don't think it's a good way forward. 

Let me think more on this. 🤔

Reply:
Further on this note I should point out that we use this "reusing the query" approach in, 

https://github.com/opencollective/opencollective-frontend/blob/a62fee101b1829e3a437bc8b90e5500f7dbe514e/components/collective-page/sections/Contribute.js#L132-L138

My suggestion is we also change that with a smaller query like we did here and in `ChangelogTrigger` which I mentioned above. Because this way it's less error prone. 🙂

Reply:
It's different to have that in the contribute section, because we know there that the component is a direct child of the collective page, thus it's fine to import the parent's query and reload it.

Here, I think a better alternative is to use `readFragment`/`writeFragment`. It's a simpler way to update a single value in the cache.

Reply:
@Betree : Thanks. I've just used the `writeFragment` and simplified the code in the `updates` function. I think now it looks better. Let me know. 🙂